layers/core_checks/cc_cmd_buffer.cpp

```diff
@@ -879,6 +879,91 @@ bool CoreChecks::PreCallValidateCmdExecuteCommands(VkCommandBuffer commandBuffer
             skip |= LogError("VUID-vkCmdExecuteCommands-pCommandBuffers-00088", objlist, cb_loc,
                              "(%s) is not VK_COMMAND_BUFFER_LEVEL_SECONDARY.", FormatHandle(pCommandBuffers[i]).c_str());
         } else {
+            if (cb_state.activeRenderPass) {
+                if (cb_state.hasRenderPassInstance && cb_state.activeRenderPass->UsesDynamicRendering() &&
+                    sub_cb_state.activeRenderPass->UsesDynamicRendering()) {
+                    const auto *location_info = vku::FindStructInPNextChain<VkRenderingAttachmentLocationInfoKHR>(
+                        sub_cb_state.activeRenderPass->inheritance_rendering_info.pNext);
+
+                    if (location_info) {
+                        const std::string vuid_090504 = "VUID-vkCmdExecuteCommands-pCommandBuffers-09504";
+                        const LogObjectList objlist(commandBuffer, pCommandBuffers[i]);
+                        skip |= ValidateRenderingAttachmentLocationsKHR(*location_info, objlist, cb_loc.dot(Field::pNext));
+
+                        if (location_info->colorAttachmentCount != cb_state.rendering_attachments.color_indexes.size()) {
+                            skip |=
+                                LogError(vuid_090504, objlist, cb_loc.dot(Field::pNext),
```

Comment:
```suggestion
                                LogError(vuid_090504, objlist, cb_loc.pNext(Struct::VkRenderingAttachmentLocationInfoKHR, Field::colorAttachmentCount), "(%" PRIu32 ") does not match the implicit ....
```

will print out a nicer error message showing where this occured

Reply:
same for all checks below